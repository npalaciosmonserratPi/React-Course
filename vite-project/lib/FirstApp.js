"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FirstApp = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const FirstApp = ({
  title,
  subTitle
}) => {
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("h1", {
      children: [title, " "]
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("h2", {
      children: subTitle
    })]
  });
};
exports.FirstApp = FirstApp;
FirstApp.defaultProps = {
  // title: 'No hay titulo',
  subTitle: 'No hay subt√≠tulo'
};
FirstApp.propTypes = {
  title: _propTypes.default.string.isRequired
};