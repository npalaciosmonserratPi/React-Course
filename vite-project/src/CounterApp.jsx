import { useState } from "react";

export const CounterApp = ({value}) => {
  const [counter, setCounter] = useState(value);
  //El estado cambiÃ³, por lo tanto se debe volver a renderizar el componente. 
  //Eso ocurre cuando se ejecuta el setCounter.

  const handleAdd = () => {
    setCounter(counter + 1);
    //setCounter((c) => {c+1})
  };

  const hanldeSubstract = () =>{
    setCounter(counter - 1);
  }

  const handleReset = () =>{
    setCounter(value);
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={hanldeSubstract}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
    </>
  );
};

