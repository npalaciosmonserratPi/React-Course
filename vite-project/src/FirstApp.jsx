import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1> 
      <h2>{ subTitle }</h2>
    </>
  );
};
 
FirstApp.defaultProps = {
 title: 'No hay titulo',
 subTitle: 'No hay subt√≠tulo'
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

