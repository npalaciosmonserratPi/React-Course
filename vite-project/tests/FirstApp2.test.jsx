import { getByText, render, screen} from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => { 

    const title = 'Hola, soy un titulo';
    const subTitle = "Hola, soy un subtitulo";

    test('debe hacer match con el snapshot',()=>{

        const {container} = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje "Hola, soy un titulo"', ()=>{

        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();

    })

    test('debe de mostrar el tÃ­tulo en un h1', ()=>{
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML ).toContain(title);
    })

    test('Verificar que existen los props del subtitulo', ()=>{
        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getByText("Hola, soy un subtitulo").innerHTML).toContain(subTitle);
        screen.debug();
    })

})