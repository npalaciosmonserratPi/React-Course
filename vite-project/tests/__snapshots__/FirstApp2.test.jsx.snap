// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy un titulo
  </h1>
  <h2>
    No hay subt√≠tulo
  </h2>
</div>
`;
